spring-resttemplate-2/src/main/java/com/baeldung/resttemplate/json/consumer/service/UserConsumerServiceImpl.java

```diff
@@ -0,0 +1,90 @@
+package com.baeldung.resttemplate.json.consumer.service;
+
+import com.baeldung.resttemplate.json.model.Address;
+import com.baeldung.resttemplate.json.model.User;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class UserConsumerServiceImpl implements UserConsumerService {
+
+    private static final String BASE_URL = "http://localhost:8080/users";
+    private final RestTemplate restTemplate;
+
+    public UserConsumerServiceImpl(RestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+    }
+
+    @Override
+    public List<String> processUserDataFromObjectArray() {
+        ObjectMapper mapper = new ObjectMapper();
+        ResponseEntity<Object[]> responseEntity = restTemplate.getForEntity(BASE_URL, Object[].class);
+        Object[] objects = responseEntity.getBody();
+        return Arrays.stream(objects)
+                       .map(object -> mapper.convertValue(object, User.class))
```

Comment:
These are very heavily indented. Please check the indentation style. I suspect they should be indented by 2 additional spaces, rather than 12.

Reply:
Quick reminder on this one.